src/qilib/data_set/mongo_data_set_io.py

```diff
@@ -0,0 +1,171 @@
+"""Quantum Inspire library
+
+Copyright 2019 QuTech Delft
+
+qilib is available under the [MIT open-source license](https://opensource.org/licenses/MIT):
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+import base64
+from typing import Optional, Dict, Any, Union, Tuple
+
+import numpy as np
+from bson import ObjectId
+from pymongo import MongoClient
+from pymongo.change_stream import CollectionChangeStream
+
+
+class DocumentNotFoundError(Exception):
+    """ Error that is raised when document is not found."""
+
+
+class NumpyKeys:
+    """The custom values types for encoding and decoding numpy arrays."""
+    OBJECT: str = '__object__'
+    CONTENT: str = '__content__'
+    DATA_TYPE: str = '__data_type__'
+    SHAPE: str = '__shape__'
+    ARRAY: str = '__ndarray__'
+
+
+class MongoDataSetIO:
+    """ Helper class for the MongoDataSetIOReader and -Writer"""
+
+    def __init__(self, name: Optional[str] = None, document_id: Optional[str] = None,
+                 create_if_not_found: Optional[bool] = True, database: str = 'qilib',
```

Comment:
It's good to see the database and collection name available as paremeters here, but now we have multiple instances of the magic constants `'qilib'` and `'data_sets'` (also in the writer), meaning that if we ever want to change the defaults we have to change it in many places. Would it be good to define these names once as constants and re-use them wherever we want the default names? This also allows someone, if they really want, to change the defaults at run-time.